src/crimson/common/config_proxy.cc

```diff
@@ -3,6 +3,20 @@
 
 #include "config_proxy.h"
 
+#if __has_include(<filesystem>)
```

Comment:
My impression is we'll need to abstract this somehow in the future.

Reply:
once we bump up the seastar to its C++17 only version, we will have the luxury dropping all these `#if` in crimson.